File: permanentPlant.js
import Employee from './employee'

class PermanentPlant extends Employee{
    constructor(name, direction, civilStatus, dateOfBirth, minimunWage, numberOfChildren, antiquity) {
        super(name, direction, civilStatus, dateOfBirth,minimunWage);
        this.numberOfChildren = numberOfChildren;
        this.antiquity = antiquity;
    }
    
    /**GETTERS */
    getNumberOfChilder(){
        return this.numberOfChildren;
    }

    getAntiquity(){
        return this.antiquity;
    }
    /**SETTERS */
    setNumberOfChilder(number){
        this.numberOfChildren = numberOfChildren;
    }

    setAtiquity(antiquity){
        this.antiquity = antiquity;
    }

    
    /**FUNCTIONS*/
    netSalary(){
        return(this.grossSalary() - this.withHoldings());
    }
    /**COMO SACO EL GET DE LA SUPER CLASE MINIMUN WAGE */
    grossSalary(){
        return(this.getMinimumWage() + this.familySalary() + this.antiquity());
    }

    familySalary(){
        return(this.childAllowance() + this.spousalAllowance())
    }

    childAllowance(){
        return(this.getNumberOfChildren() * 150);
    }

    spousalAllowance(){
        if(this.getCivilSatus() === 'Married'){
            return 100;
        }
        else{
            return 0;
        }
    }

    antiquity(){
        return(this.getAntiquity() * 50);
    }

    withHoldings(){
        return(this.prepairdHealth() + this.retirementContribution());
    }

    prepairdHealth(){
        return((this.grossSalary()*10)/100 + (this.getNumberOfChildren() * 20) );
    }

    retirementContribution(){
        return((this.grossSalary()*15)/100);
    }
}

File: TP02 - vJAVA/src/unq/PermanentPlant.java
package unq;

public class PermanentPlant extends Employee {
	private Integer numberOfChildren;
	private Integer Antiquity;

	public PermanentPlant(String name, Integer age, String direction, String civilStatus, String dateOfBirth,
			Integer minimunWage, Integer numberOfChildren, Integer Antiquity) {
		super(name, age, direction, civilStatus, dateOfBirth, minimunWage);
		this.numberOfChildren = numberOfChildren;
		this.Antiquity = Antiquity;
		
	}

	public Integer getNumberOfChildren() {
		return numberOfChildren;
	}

	public void setNumberOfChildren(Integer numberOfChildren) {
		this.numberOfChildren = numberOfChildren;
	}

	public Integer getAntiquity() {
		return Antiquity;
	}

	public void setAntiquity(Integer antiquity) {
		Antiquity = antiquity;
	}

	@Override
	public Integer grossSalary() {
		return this.getMinimunWage() + this.familySalary() + this.antiquity();
	}

	private Integer antiquity() {
		return this.getAntiquity() * 50;
	}

	private Integer familySalary() {
		return this.childAllowance() + this.spoualAllowance();
	}

	private int spoualAllowance() {
		Integer res = 0; 
		if (this.getCivilStatus() == "Married") {
			res = 100;
		}
		return res;
	}

	private int childAllowance() {
		return this.getNumberOfChildren() * 150;
	}

	@Override
	public Integer netSalary() {
		return this.grossSalary() - this.withHoldings();
	}

	@Override
	public Integer withHoldings() {
		return this.healthInsurance() + this.retirementContribution();
	}

	@Override
	public Integer retirementContribution() {
		return (this.grossSalary() * 15) / 100;
	}

	@Override
	public Integer healthInsurance() {
		return (((this.grossSalary() * 10) /100 + (this.getNumberOfChildren() * 20)));
	}
	
	
}
